odoo/modules/db.py

```diff
@@ -138,14 +138,29 @@ def create_categories(cr, categories):
     return p_id
 
 def has_unaccent(cr):
-    """ Test if the database has an unaccent function.
+    """ Test if the database has an unaccent function and returning it `provolatile` if it exists.
 
     The unaccent is supposed to be provided by the PostgreSQL unaccent contrib
     module but any similar function will be picked by OpenERP.
 
+    The `provolatile` of unaccent allows to know if the unaccent function
+    can be used to create index (it should be 'i' - means immutable), see
+    https://www.postgresql.org/docs/current/catalog-pg-proc.html.
+
+    :return: False if unaccent doesn't exist, `provolatile` of the unaccent method otherwise (always truly)
     """
-    cr.execute("SELECT proname FROM pg_proc WHERE proname='unaccent'")
-    return len(cr.fetchall()) > 0
+    cr.execute("""
+        SELECT provolatile
+        FROM pg_proc p
+            LEFT JOIN pg_catalog.pg_namespace ns ON p.pronamespace = ns.oid
+        WHERE proname = 'unaccent'
+              AND pronargs = 1
+              AND ns.nspname = 'public'
+    """)
+    result = cr.fetchall()
```

Comment:
If you're going to rewrite everything, use `fetchone` and check for `None`?

Reply:
Yep, it seems legit.